codebase/src/graph/graphAnimation.ts

```diff
@@ -1,20 +1,42 @@
-import { type GNode, type GEdge, type AdjacencyList, Status } from "./types";
+import { type GNode, type GEdge } from "./types";
+import { delay } from "@/sorting/algorithms/bubble";
+import { drawEdges } from "@/components/edge";
+import { drawNodes } from "@/components/node";
 
-const animateGraph = (nodes: GNode[], edges: GEdge[], trace: number[][]) => {
+export const animateGraph = async (
+  nodes: GNode[],
+  edges: GEdge[],
+  trace: (number | null)[][],
+  ctx: CanvasRenderingContext2D,
+  nodeRadius: number,
+  speed: number
```

Comment:
what does speed mean. speed until next step of the trace in the visualization? what unit is speed?